trie/child_storage.go

```diff
@@ -0,0 +1,84 @@
+package trie
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/common"
+)
+
+// ChildStorageKeyPrefix is the prefix for all child storage keys
+var ChildStorageKeyPrefix = []byte(":child_storage:")
+
+// PutChild inserts a child trie into the main trie at key :child_storage:[keyToChild]
+func (t *Trie) PutChild(keyToChild []byte, child *Trie) error {
+	childHash, err := child.Hash()
+	if err != nil {
+		return err
+	}
+
+	key := append(ChildStorageKeyPrefix, keyToChild[:]...)
```

Comment:
Not necessary to convert a slice to a slice here with `keyToChild[:]...`

Reply:
fixed! thanks